java-spiffe-provider/src/main/java/io/spiffe/provider/SpiffeTrustManager.java

```diff
@@ -28,6 +29,7 @@
 
     private final BundleSource<X509Bundle> x509BundleSource;
     private final Supplier<Set<SpiffeId>> acceptedSpiffeIdsSupplier;
+    private final SpiffeIdVerifier spiffeIdVerifier;
```

Comment:
I think it would make the logic much simpler if we used only one mechanism for verifying SPIFFE IDs in this class. What if rather than this class using either `acceptedSpiffeIdsSupplier` or `spiffeIdVerifier` to determine which form of validation to perform, it instead relied on a default implementation of `SpiffeIdVerifier` that uses the `acceptedSpiffeIdsSupplier`? e.g.

```java
public final class StaticSpiffeIdVerifier {
    private final Supplier<Set<SpiffeId>> acceptedSpiffeIdsSupplier;
    public StaticSpiffeIdVerifier(@NonNull final Supplier<Set<SpiffeId>> acceptedSpiffeIdsSupplier) {
        this.acceptedSpiffeIdsSupplier = acceptedSpiffeIdsSupplier;
    }

    public void verify(SpiffeId spiffeId, X509Certificate[] verifiedChain) throws SpiffeIdVerificationException {
        X509SvidValidator.verifySpiffeId(chain[0], acceptedSpiffeIdsSupplier);
    }
}

public final class SpiffeTrustManager extends X509ExtendedTrustManager {
    private final SpiffeIdVerifier spiffeIdVerifier;

    public SpiffeTrustManager(@NonNull final BundleSource<X509Bundle> x509BundleSource,
                              @NonNull final Supplier<Set<SpiffeId>> acceptedSpiffeIdsSupplier) {
        this.spiffeIdVerifier = new StaticSpiffeIdVerifier(acceptedSpiffeIdsSupplier);
    }
}
```